openmetadata-ui/src/main/resources/ui/src/utils/SearchClassBase.ts

```diff
@@ -344,6 +345,18 @@ class SearchClassBase {
       });
     }
 
+    if (entity.entityType === EntityType.CHART) {
+      const dashboard = (entity as Chart).dashboards?.[0];
```

Comment:
Hey @Ashish8689, why are we only grabbing the first element from the dashboard list?

Reply:
so, the list contains the dashboard which has this chart associated with it. And since we don't have any page dedicated to the chart we are redirecting it to the dashboard where it is being used. and so we have used the first dashboard to redirect